### DISCUSSION TITLE
Test setup and teardown

### DISCUSSION THREAD

**Discussion by orbiteleven** [mobile-dev-inc/Maestro#1082] [Q&A] [answered]
What are folks doing for test setup and teardown. As a use case, I'd like to have a test of my signup/login flow where I have:
- An existing user for which one cannot create a new account using the same email
- A new user created with a new email address
- This new user needs to verify their email to continue

We're currently using Firestore and I'm thinking an HTTPS Functions endpoint that could be hit on test startup would be sufficient, but it seems like Rhino is incredibly hobbled in its ability to send HTTPS requests.

I've looked at `openLink`, but that feels even more hacky and doesn't give us a good way (AFAICT) to do even basic shared token auth.

How are others handling this?

**Comment by orbiteleven** [comment]
https://github.com/mobile-dev-inc/maestro/issues/1081 Is somewhat related

**Comment by orbiteleven** [comment, marked as answer]
Oh... and RTFM, I just came across this, which seems to do the trick: https://maestro.mobile.dev/advanced/javascript/make-http-s-requests

**Comment by axelniklasson** [comment]
Closing this out as this was resolved - glad you found a workaround @orbiteleven!
